tensorflow/contrib/lite/kernels/gather_test.cc

```diff
@@ -76,6 +76,27 @@ TEST(GatherOpTest, Shuffle) {
               ElementsAreArray(ArrayFloatNear({0.7, 0.8, -2, 0.2})));
 }
 
+TEST(GatherOpTest, Index) {
+  GatherOpModel m({2, 2}, TensorType_FLOAT32, {});
+  m.SetInputFloat({-2.0, 0.2, 0.7, 0.8});
+  m.SetPositions({1});
+  m.Invoke();
+  EXPECT_THAT(m.GetOutputFloat(),
+              ElementsAreArray(ArrayFloatNear({0.7, 0.8})));
+  EXPECT_THAT(m.GetOutputShape(),
+              ElementsAreArray({2}));
+}
+
+TEST(GatherOpTest, IndexToCornerCase) {
```

Comment:
Why is this a corner case? Getting the 2nd element from a 1D tensor with 3 elements seems pretty normal to me.

Reply:
I suppose it is corner since the most natural way (at least in Python API) of getting an element from 1D tensor is through **strided_slice**.

Unfortunately getting **strided_slice** ready is more difficult than patching **gather**. In fact I also wrote a *toco* transformation to convert this kind of **strided_slice** into 0D **gather**.

Reply:
Another aspect is that 0D tensor is a special case in current TFLite - many code cannot distinguish 0D and 1D tensor, and some workarounds are needed.

Reply:
Thanks for your explanation. I see what you meant. 
On the other hand, it's a bit hard to understand what this means without thinking about `strided_slice`, and without extra comments. 
Do you think it's better to call this something like "GatherFrom1DTensor"?